src/pkg/variables/common.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package variables contains functions for interacting with variables
+package variables
+
+// VariableConfig represents a value to be templated into a text file.
+type VariableConfig struct {
+	templatePrefix string
+	deprecatedKeys map[string]string
+
+	ApplicationTemplates map[string]*TextTemplate
+	SetVariableMap       SetVariableMap
+	Constants            []Constant
+
+	logger func(format string, a ...any)
```

Comment:
I'm fine with this being in a separate PR, but would like to see this follow the slog.Logger interface eventually. Our implementation of slogger needs some love too, though it is functional. Unless you disagree I'll make an issue to track

Reply:
implemented